modules/gcp/region.go

```diff
@@ -100,10 +101,19 @@ func GetRandomZoneE(t *testing.T, projectID string, approvedZones []string, forb
 	}
 
 	zonesToPickFrom = collections.ListSubtract(zonesToPickFrom, forbiddenZones)
-	zone := random.RandomString(zonesToPickFrom)
 
-	logger.Logf(t, "Using Zone %s", zone)
-	return zone, nil
+	zone := ""
+	maxAttempts := 1000
+
+	for i := 0; i < maxAttempts; i++ {
+		zone = random.RandomString(zonesToPickFrom)
+		if !isInRegions(zone, forbiddenRegions) {
```

Comment:
Is it possible to just filter out `forbiddenRegions` from `zonesToPickFrom` directly before randomly picking? I.e create a new list that only includes those that are not forbidden? Similar to how we do it [for aws](https://github.com/gruntwork-io/terratest/blob/master/modules/aws/region.go#L54).